src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/reporting/ndmis/performance/NdmisPerformanceReportJobConfiguration.kt

```diff
@@ -161,16 +168,30 @@ class NdmisPerformanceReportJobConfiguration(
       .reader(ndmisReader)
       .processor(processor)
       .writer(writer)
+      .faultTolerant()
+      .skipPolicy(skipPolicy)
       .build()
   }
 
   private val pushToS3Step = stepBuilderFactory["pushToS3Step"]
     .tasklet { _: StepContribution, _: ChunkContext ->
       listOf(referralReportPath, complexityReportPath, appointmentsReportPath).forEach { path ->
         s3Service.publishFileToS3(ndmisS3Bucket, path, "export/csv/reports/")
-        File(path.toString()).delete()
+//        File(path.toString()).delete()
       }
       RepeatStatus.FINISHED
     }
     .build()
 }
+
+class NPESkipPolicy : SkipPolicy {
+  override fun shouldSkip(t: Throwable, skipCount: Int): Boolean {
+    return when (t) {
+      is NullPointerException -> {
+        NdmisPerformanceReportJobConfiguration.logger.warn("skipping referral with unexpected state", t)
```

Comment:
nit: since this class is defined on its own, it's generic in terms of what it's skipping. `skipping referrral...` may not always be correct.